bundles/com.espressif.idf.launch.serial.ui/src/com/espressif/idf/launch/serial/ui/internal/CMakeMainTab2.java

```diff
@@ -74,48 +77,229 @@
+		GridLayout layout = new GridLayout();
+		layout.numColumns = 1;
+		GridData defaultCompositeGridData = new GridData(GridData.FILL_HORIZONTAL);
+		defaultCompositeGridData.exclude = true;
+		switchGridDatas.put(FlashInterface.UART, defaultCompositeGridData);
+		defaultComposite.setLayout(layout);
+		defaultComposite.setLayoutData(defaultCompositeGridData);
+
+		createLocationComponent(defaultComposite);
+		createWorkDirectoryComponent(defaultComposite);
+
+		GridData locationAndWorkDirGroupData = new GridData(SWT.FILL, SWT.NONE, true, false);
 		locationField.getParent().setLayoutData(locationAndWorkDirGroupData);
 		workDirectoryField.getParent().setLayoutData(locationAndWorkDirGroupData);
+
+		uartAgrumentsField = new Text(parent, SWT.MULTI | SWT.WRAP | SWT.BORDER | SWT.V_SCROLL);
+
+		createArgumentComponent(defaultComposite, uartAgrumentsField);
+		createVerticalSpacer(defaultComposite, 1);
+	}
+
+	protected void createJtagflashComposite(Composite parent) {
+
+		Composite jtagComposite = new Composite(parent, SWT.NONE);
+		switchComposites.put(FlashInterface.JTAG, jtagComposite);
+		jtagComposite.setFont(parent.getFont());
+		GridLayout layout = new GridLayout();
+		layout.numColumns = 1;
+		GridData jtagCompositeGridData = new GridData(GridData.FILL_HORIZONTAL);
+		jtagCompositeGridData.exclude = true;
+		switchGridDatas.put(FlashInterface.JTAG, jtagCompositeGridData);
+		jtagComposite.setLayout(layout);
+		jtagComposite.setLayoutData(jtagCompositeGridData);
+
+		String selectedTarget = getLaunchTarget();
+		EspConfigParser parser = new EspConfigParser();
+		createOpenOcdSetupComponent(jtagComposite, selectedTarget, parser);
+
+		jtagArgumentsField = new Text(parent, SWT.MULTI | SWT.WRAP | SWT.BORDER | SWT.V_SCROLL);
+		createArgumentComponent(jtagComposite, jtagArgumentsField);
+		createVerticalSpacer(jtagComposite, 1);
+	}
+
+	protected void createDfuComposite(Composite parent) {
+		Composite dfuComposite = new Composite(parent, SWT.NONE);
+		switchComposites.put(FlashInterface.DFU, dfuComposite);
+		dfuComposite.setFont(parent.getFont());
+		GridLayout layout = new GridLayout();
+		layout.numColumns = 1;
+		GridData dfuCompositeGridData = new GridData(GridData.FILL_HORIZONTAL);
+		dfuCompositeGridData.exclude = true;
+		switchGridDatas.put(FlashInterface.DFU, dfuCompositeGridData);
+		dfuComposite.setLayout(layout);
+		dfuComposite.setLayoutData(dfuCompositeGridData);
+
+		Composite targetComposite = new Composite(dfuComposite, SWT.NONE);
+		layout = new GridLayout();
+		layout.numColumns = 3;
+		GridData targetGridData = new GridData(GridData.FILL_HORIZONTAL);
+		targetComposite.setLayout(layout);
+		targetComposite.setData(targetGridData);
+
+		Label comboTargetLbl = new Label(targetComposite, SWT.NONE);
+		comboTargetLbl.setText(Messages.CMakeMainTab2_TargetsComboLbl);
+		comboTargets = new Combo(targetComposite, SWT.DROP_DOWN | SWT.READ_ONLY);
+		comboTargets.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, false, false));
```

Comment:
Can we try to avoid SWT.CENTER? This has previously caused issues when rendering SWT controls on Linux. I faced the issue with Install Tools Wizard.

Reply:
Hi @alirana01, thank you for reviewing. Fixed